pkg/core/transaction/transaction.go

```diff
@@ -297,6 +290,15 @@ func (t *Transaction) FeePerByte() int64 {
 	return t.feePerByte
 }
 
+// Sender returns the sender of the transaction which is always on the first place
+// in the transaction's signers list.
+func (t *Transaction) Sender() util.Uint160 {
+	if len(t.Signers) == 0 {
+		panic("transaction do not have signers")
```

Comment:
If we panic here, we should check if `Signers` array is non-empty in `UnmarshalJSON`.

Reply:
Fixed in a separate commit. Now all checks from transaction.decodeHashableFields are also performed in UnmarshalJSON.